security/kerberos_pac.ksy

```diff
@@ -0,0 +1,542 @@
+      - id: name_length
+        -orig-id: NameLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the Name field
+      - id: name
+        -orig-id: Name
+        type: str
+        size: name_length
+        encoding: UTF-16LE
+        doc: An array of 16-bit Unicode characters in little-endian format that contains the client's account name
+  upn_dns_info:
+    doc: "See [MS-PAC] 2.10 UPN_DNS_INFO"
+    seq:
+      - id: upn_length
+        -orig-id: UpnLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the UPN information
+      - id: upn_offset
+        -orig-id: UpnOffset
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that contains the offset to the beginning of the UPN information, in bytes, from the beginning of the UPN_DNS_INFO structure
+      - id: dns_domain_name_length
+        -orig-id: DnsDomainNameLength
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that specifies the length, in bytes, of the DNS information
+      - id: dns_domain_name_offset
+        -orig-id: DnsDomainNameOffset
+        type: u2
+        doc: An unsigned 16-bit integer in little-endian format that contains the offset to the beginning of the DNS information, in bytes, from the beginning of the UPN_DNS_INFO structure
+      - id: padding1
+        size: upn_offset - 8
+      - id: upn
+        type: str
+        size: upn_length
+        encoding: UTF-16LE
+      - id: padding2
+        size: dns_domain_name_offset - upn_offset - upn_length
+      - id: dns_domain_name
+        type: str
+        size: dns_domain_name_length
+        encoding: UTF-16LE
+  pac_signature_data:
+    doc: "See [MS-PAC] 2.8 PAC_SIGNATURE_DATA"
+    seq:
+      - id: signature_type
+        -orig-id: SignatureType
+        type: u4
+        enum: signature_type_enum
+        doc: A 32-bit unsigned integer value in little-endian format that defines the cryptographic system used to calculate the checksum
+      - id: signature_16
+        -orig-id: Signature
+        type: u1
+        repeat: expr
+        repeat-expr: 16
+        if: 'signature_type == signature_type_enum::kerb_checksum_hmac_md5'
+      - id: signature_12
+        -orig-id: Signature
+        type: u1
+        repeat: expr
+        repeat-expr: 12
+        if: '(signature_type == signature_type_enum::hmac_sha1_96_aes128) or (signature_type == signature_type_enum::hmac_sha1_96_aes256)'
+      - id: ro_dc_identifier
+        -orig-id: RODCIdentifier
+        type: u2
+        repeat: eos
+        doc: A 16-bit unsigned integer value in little-endian format that contains the first 16 bits of the key version number ([MS-KILE] section 3.1.5.8) when the KDC is an RODC
```

Comment:
According to `doc`, didn't you mean this instead?

```suggestion
      - id: ro_dc_identifier
        -orig-id: RODCIdentifier
        type: u2
        if: not _io.eof
        doc: A 16-bit unsigned integer value in little-endian format that contains the first 16 bits of the key version number ([MS-KILE] section 3.1.5.8) when the KDC is an RODC
```

It should not be an array, if I understand correctly.